---
title: "PD_omics_review"
author: "<h3>Author</h3>Brian M. Schilder"
date: "`r Sys.Date()`"
output: rmarkdown::github_document
vignette: >
  %\VignetteIndexEntry{MungeSumstats}
  %\VignetteEngine{knitr::rmarkdown}
  %\usepackage[utf8]{inputenc}    
---


Data and code associated with the Parkinson's Disease review paper by Schilder et al. (2021).

If you use the data in this repository please cite both the original authors of the data, as well as:

> Brian M. Schilder, Elisa Navarro, Towfique Raj (2021) *Multi-omic insights into Parkinsonâ€™s diseases: from genetic associations to functional mechanisms* 

## Abstract 

> Genome-Wide Association Studies (GWAS) have elucidated the genetic components of Parkinson's Disease (PD). However,  the vast majority of GWAS association signals  fall within non-coding regions, translating these results into an interpretable, mechanistic understanding of the disease etiology remains a major challenge in the field. In this review, we provide an overview of the approaches to prioritize putative causal variants and genes as well as summarise the primary findings of previous studies. We then discuss recent progress made in efforts to integrate multi-omics data to identify likely pathogenic cell types and biological pathways implicated in PD pathogenesis. We have compiled complete PD GWAS cell-type and tissue enrichment summary statistics from multiple studies and provided them in a standardized format as a resource for the research community to use in meta-analyses. Additionally, we discuss experimental approaches that will be needed to further elucidate the effects of functional variants and genes on cellular phenotypes and disease risk.


## Scripts 

### [metaanalysis_preprocess](https://rajlabmssm.github.io/PD_omics_review/code/metaanalysis_preprocess.html)

Code to preprocess the meta-analysis datasets (**Table S1**).


### [metaanalysis_merge](https://rajlabmssm.github.io/PD_omics_review/code/metaanalysis_merge.html) 

Code to harmonize all cell-type/tissue meta-analysis datasets and generate **Figure 2**.


### [genetic_correlations](https://rajlabmssm.github.io/PD_omics_review/code/genetic_correlations.html) 

Exploratory plots of cross-phenotype genetic correlations datasets (**Table S2**).


### [geneshot](https://rajlabmssm.github.io/PD_omics_review/code/geneshot.html)

Code to generate UMAP plot with [Geneshot](https://maayanlab.cloud/geneshot/) query results of "Parkinson's" (**Figure S1**).


### [multimodal_networks](https://rajlabmssm.github.io/PD_omics_review/code/multimodal_networks.html)

Exploratory analyses of multi-modal networks using genes/celltypes/tissue/pathways associated with Parkinson's Disease.


## Plots

High-resolution plots generated by these scripts can be found [here](https://github.com/RajLabMSSM/PD_omics_review/tree/main/plots). 


## Session info 
 
<details>

```{r}
utils::sessionInfo()
```

</details> 
