PD\_omics\_review
================
<h3>
Author
</h3>
Brian M. Schilder

2021-08-22

Data and code associated with the Parkinson’s Disease review paper by
Schilder, Navarro & Raj (2021).

If you use the data in this repository please cite both the original
authors of the data, as well as:

> Brian M. Schilder, Elisa Navarro, Towfique Raj (2021) *Multi-omic
> insights into Parkinson’s diseases: from genetic associations to
> functional mechanisms*

## Abstract

> Genome-Wide Association Studies (GWAS) have elucidated the genetic
> components of Parkinson’s Disease (PD). However, the vast majority of
> GWAS association signals fall within non-coding regions, translating
> these results into an interpretable, mechanistic understanding of the
> disease etiology remains a major challenge in the field. In this
> review, we provide an overview of the approaches to prioritize
> putative causal variants and genes as well as summarise the primary
> findings of previous studies. We then discuss recent progress made in
> efforts to integrate multi-omics data to identify likely pathogenic
> cell types and biological pathways implicated in PD pathogenesis. We
> have compiled complete PD GWAS cell-type and tissue enrichment summary
> statistics from multiple studies and provided them in a standardized
> format as a resource for the research community to use in
> meta-analyses. Additionally, we discuss experimental approaches that
> will be needed to further elucidate the effects of functional variants
> and genes on cellular phenotypes and disease risk.

## Scripts

### [metaanalysis\_preprocess](https://rajlabmssm.github.io/PD_omics_review/code/metaanalysis_preprocess.html)

Code to preprocess the meta-analysis datasets (**Table S1**).

### [metaanalysis\_merge](https://rajlabmssm.github.io/PD_omics_review/code/metaanalysis_merge.html)

Code to harmonize all cell-type/tissue meta-analysis datasets and
generate **Figure 2**.

### [genetic\_correlations](https://rajlabmssm.github.io/PD_omics_review/code/genetic_correlations.html)

Exploratory plots of cross-phenotype genetic correlations datasets
(**Table S2**).

### [geneshot](https://rajlabmssm.github.io/PD_omics_review/code/geneshot.html)

Code to generate UMAP plot with
[Geneshot](https://maayanlab.cloud/geneshot/) query results of
“Parkinson’s” (**Figure S1**).

### [multimodal\_networks](https://rajlabmssm.github.io/PD_omics_review/code/multimodal_networks.html)

Exploratory analyses of multi-modal networks using
genes/celltypes/tissue/pathways associated with Parkinson’s Disease.

## Plots

High-resolution plots generated by these scripts can be found
[here](https://github.com/RajLabMSSM/PD_omics_review/tree/main/plots).

## Session info

<details>

``` r
utils::sessionInfo()
```

    ## R version 4.1.0 (2021-05-18)
    ## Platform: x86_64-pc-linux-gnu (64-bit)
    ## Running under: Ubuntu 20.04.2 LTS
    ## 
    ## Matrix products: default
    ## BLAS/LAPACK: /usr/lib/x86_64-linux-gnu/openblas-pthread/libopenblasp-r0.3.8.so
    ## 
    ## locale:
    ##  [1] LC_CTYPE=en_US.UTF-8       LC_NUMERIC=C              
    ##  [3] LC_TIME=en_US.UTF-8        LC_COLLATE=en_US.UTF-8    
    ##  [5] LC_MONETARY=en_US.UTF-8    LC_MESSAGES=C             
    ##  [7] LC_PAPER=en_US.UTF-8       LC_NAME=C                 
    ##  [9] LC_ADDRESS=C               LC_TELEPHONE=C            
    ## [11] LC_MEASUREMENT=en_US.UTF-8 LC_IDENTIFICATION=C       
    ## 
    ## attached base packages:
    ## [1] stats     graphics  grDevices utils     datasets  methods   base     
    ## 
    ## loaded via a namespace (and not attached):
    ##  [1] compiler_4.1.0    magrittr_2.0.1    tools_4.1.0       htmltools_0.5.1.1
    ##  [5] yaml_2.2.1        stringi_1.7.3     rmarkdown_2.10    knitr_1.33       
    ##  [9] stringr_1.4.0     xfun_0.25         digest_0.6.27     rlang_0.4.11     
    ## [13] evaluate_0.14

</details>
